minecode/visitors/npm.py

```
@@ -103,3 +106,73 @@ def get_uris(self, content):
                 data=json.dumps(doc, separators=(',', ':'), ensure_ascii=False),
                 # note: visited is True since there nothing more to visit
                 visited=True)
+
+def get_package_json(namespace, name, version):
+    """
+    Return the contents of the package.json file of the package described by the purl
+    field arguments in a string.
+    """
+    # Create URLs using purl fields
+    url = npm_api_url(
+        namespace=namespace,
+        name=name,
+        version=version,
+    )
+
+    try:
+        response = requests.get(url)
+        response.raise_for_status() 
+        return response.json()
+    except requests.exceptions.HTTPError as err:
+        print(f"HTTP error occurred: {err}")
+
+
+def map_npm_package(package_url):
+    """
+    Add a npm `package_url` to the PackageDB.
+
+    Return an error string if any errors are encountered during the process
+    """
+    from minecode.management.commands.priority_queue import add_package_to_scan_queue
+    from minecode.management.commands.run_map import merge_or_create_package
+
+    pacakge_json = get_package_json(
```

I think `get_package_json` should be wrapped in a try-except block in the case where the request fails, we can return the request error message.

---

Already handling request error inside `get_package_json`.
Do we still want to wrap it inside try-except block?
https://github.com/nexB/purldb/blob/4ce38fb525de74891f27b4df21deb2ad27c20e73/minecode/visitors/npm.py#L121-L128

---

I think you're good. I misred the code.